pkg/util/fsnotification/fsnotification.go

```diff
@@ -0,0 +1,52 @@
+package fsnotification
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/glog"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+// AddRecursiveWatch handles adding watches recursively for the path provided
+// and its subdirectories.
+// recursive logic from https://github.com/bronze1man/kmg/blob/master/fsnotify/Watcher.go
+func AddRecursiveWatch(watcher *fsnotify.Watcher, path string) error {
+	file, err := os.Stat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("error introspecting path %s: %v", path, err)
+	}
+	if !file.IsDir() {
+		return nil
+	}
+
+	folders, err := getSubFolders(path)
+	for _, v := range folders {
+		glog.V(5).Infof("adding watch on path %s", v)
+		err = watcher.Add(v)
+		if err != nil {
+			return fmt.Errorf("error adding watcher for path %s: %v", v, err)
+		}
+	}
+	return nil
+}
+
+func getSubFolders(path string) (paths []string, err error) {
+	err = filepath.Walk(path, func(newPath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !info.IsDir() {
```

Comment:
Since we return nil in both cases, this would be easier to read like this:

```go
		if info.IsDir() {
			paths = append(paths, newPath)
		}
		return nil
```

WDYT?